--- ./original/jstree.js	2019-08-29 11:44:38.865888700 +0400
+++ jstree.js	2019-08-29 11:46:12.694171600 +0400
@@ -593,7 +593,7 @@
 					return this.nodeType === 3 && (!this.nodeValue || /^\s+$/.test(this.nodeValue));
 				})
 				.remove();
-			this.element.html("<"+"ul class='jstree-container-ul jstree-children' role='group'><"+"li id='j"+this._id+"_loading' class='jstree-initial-node jstree-loading jstree-leaf jstree-last' role='tree-item'><i class='jstree-icon jstree-ocl'></i><"+"a class='jstree-anchor' href='#'><i class='jstree-icon jstree-themeicon-hidden'></i>" + this.get_string("Loading ...") + "</a></li></ul>");
+			this.element.html("<"+"ul class='jstree-container-ul jstree-children' role='group'><"+"li id='j"+this._id+"_loading' class='jstree-initial-node jstree-loading jstree-leaf jstree-last' role='tree-item'><i class='jstree-icon jstree-ocl'></i><"+"a class='jstree-anchor'><i class='jstree-icon jstree-themeicon-hidden'></i>" + this.get_string("Loading ...") + "</a></li></ul>");
 			this.element.attr('aria-activedescendant','j' + this._id + '_loading');
 			this._data.core.li_height = this.get_container_ul().children("li").first().outerHeight() || 24;
 			this._data.core.node = this._create_prototype_node();
@@ -641,7 +641,7 @@
 			_node.appendChild(_temp1);
 			_temp1 = document.createElement('A');
 			_temp1.className = 'jstree-anchor';
-			_temp1.setAttribute('href','#');
+//			_temp1.setAttribute('href','#');
 			_temp1.setAttribute('tabindex','-1');
 			_temp2 = document.createElement('I');
 			_temp2.className = 'jstree-icon jstree-themeicon';
@@ -1629,6 +1629,7 @@
 									id			: tid,
 									text		: d.text || '',
 									icon		: d.icon !== undefined ? d.icon : true,
+                                    fallbackIcon: d.fallbackIcon,
 									parent		: p,
 									parents		: ps,
 									children	: d.children || [],
@@ -1636,7 +1637,7 @@
 									data		: d.data,
 									state		: { },
 									li_attr		: { id : false },
-									a_attr		: { href : '#' },
+									a_attr		: { /*href : '#'*/ },
 									original	: false
 								};
 							for(i in df) {
@@ -1726,7 +1727,7 @@
 								data		: null,
 								state		: { },
 								li_attr		: { id : false },
-								a_attr		: { href : '#' },
+								a_attr		: { /*href : '#'*/ },
 								original	: false
 							};
 							for(i in df) {
@@ -2012,7 +2013,7 @@
 					data		: null,
 					state		: { },
 					li_attr		: { id : false },
-					a_attr		: { href : '#' },
+					a_attr		: { /*href : '#'*/ },
 					original	: false
 				}, i, tmp, tid;
 			for(i in this._model.default_state) {
@@ -2127,7 +2128,7 @@
 					data		: d.data,
 					state		: { },
 					li_attr		: { id : false },
-					a_attr		: { href : '#' },
+					a_attr		: { /*href : '#'*/ },
 					original	: false
 				};
 			for(i in df) {
@@ -2226,7 +2227,7 @@
 				data		: null,
 				state		: { },
 				li_attr		: { id : false },
-				a_attr		: { href : '#' },
+				a_attr		: { /*href : '#'*/ },
 				original	: false
 			};
 			for(i in df) {
@@ -2538,13 +2539,13 @@
 				if(obj.icon === false) {
 					node.childNodes[1].childNodes[0].className += ' jstree-themeicon-hidden';
 				}
-				else if(obj.icon.indexOf('/') === -1 && obj.icon.indexOf('.') === -1) {
+				else if(obj.icon.indexOf('/') === -1 && obj.icon.indexOf('.') === -1 && !obj.icon.startsWith("var(")) {
 					node.childNodes[1].childNodes[0].className += ' ' + obj.icon + ' jstree-themeicon-custom';
 				}
 				else {
-					node.childNodes[1].childNodes[0].style.backgroundImage = 'url("'+obj.icon+'")';
+    				node.childNodes[1].childNodes[0].style.backgroundImage = (obj.icon.startsWith("var(")? obj.icon: 'url("' + obj.icon + '")') + (obj.fallbackIcon? ', ' + obj.fallbackIcon: '');
 					node.childNodes[1].childNodes[0].style.backgroundPosition = 'center center';
-					node.childNodes[1].childNodes[0].style.backgroundSize = 'auto';
+					node.childNodes[1].childNodes[0].style.backgroundSize = '16px 16px';
 					node.childNodes[1].childNodes[0].className += ' jstree-themeicon-custom';
 				}
 			}
@@ -3584,7 +3585,7 @@
 
 			var c = this.get_container_ul()[0].className;
 			if(!skip_loading) {
-				this.element.html("<"+"ul class='"+c+"' role='group'><"+"li class='jstree-initial-node jstree-loading jstree-leaf jstree-last' role='treeitem' id='j"+this._id+"_loading'><i class='jstree-icon jstree-ocl'></i><"+"a class='jstree-anchor' href='#'><i class='jstree-icon jstree-themeicon-hidden'></i>" + this.get_string("Loading ...") + "</a></li></ul>");
+				this.element.html("<"+"ul class='"+c+"' role='group'><"+"li class='jstree-initial-node jstree-loading jstree-leaf jstree-last' role='treeitem' id='j"+this._id+"_loading'><i class='jstree-icon jstree-ocl'></i><"+"a class='jstree-anchor'><i class='jstree-icon jstree-themeicon-hidden'></i>" + this.get_string("Loading ...") + "</a></li></ul>");
 				this.element.attr('aria-activedescendant','j'+this._id+'_loading');
 			}
 			this.load_node($.jstree.root, function (o, s) {
@@ -4802,14 +4803,17 @@
 				dom.removeClass('jstree-themeicon-custom ' + old).css("background","").removeAttr("rel");
 				if(old === false) { this.show_icon(obj); }
 			}
-			else if(icon.indexOf("/") === -1 && icon.indexOf(".") === -1) {
+    		else if(icon.indexOf("/") === -1 && icon.indexOf(".") === -1 && !obj.icon.startsWith("var(")) {
 				dom.removeClass(old).css("background","");
 				dom.addClass(icon + ' jstree-themeicon-custom').attr("rel",icon);
 				if(old === false) { this.show_icon(obj); }
 			}
 			else {
 				dom.removeClass(old).css("background","");
-				dom.addClass('jstree-themeicon-custom').css("background", "url('" + icon + "') center center no-repeat").attr("rel",icon);
+				dom.addClass('jstree-themeicon-custom').css("background", 
+                    (icon.startsWith("var(")? icon: 'url("' + icon + '")') + " center center no-repeat")
+                    .css("background-size", "16px 16px")
+                    .attr("rel",icon);
 				if(old === false) { this.show_icon(obj); }
 			}
 			return true;
